calblueprint/ogsc
# User Profile and Editing Users

# User Dashboard User View/Edit

- Can click into an individual user from the admin user dashboard
- Can edit a user's profile information, and have it reflect in the database

Note: Does not have functionality to edit user role right now - have not figured out how to do this yet without the viewing permissions information or a separate role field in the users table

## TODO (if applicable)

- [ ] Styling (the modal is not perfect right now)

## How to Test/Use PR feature

1. run `yarn dev`
2. go to `[localhost...]/admin/users`
3. click on any user - this should bring you to the user's profile page
4. click on the edit button - a form should pop up
5. fill out the form with updated information, and save - this should update the information on the user's profile  (NOTE: you might have to refresh the user profile to see changes after you click save)

## PR Checklist

Does your branch (check if true):
[X] work when you run `yarn build`?
[X] successfully run `yarn:db-migrate up` without yielding any errors?
[X] successfully run `yarn prisma introspect` without yielding any errors?
[X] successfully run `yarn dev` and support all changes that your branch intends to perform?

## Related PRs

N/A

## Migrations

N/A
## Screenshots

Users Dashboard

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/65790318/99139692-e5ef2a80-25ef-11eb-9186-5b51f9c594c0.png">

User Profile

<img width="1035" alt="image" src="https://user-images.githubusercontent.com/65790318/99139702-fef7db80-25ef-11eb-94ef-43c731f3185a.png">

Edit Form (Before Editing)

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/65790318/99139717-133bd880-25f0-11eb-989c-fbe068575d6e.png">

Edit Form (after editing)
<img width="888" alt="image" src="https://user-images.githubusercontent.com/65790318/99160688-a7647900-269e-11eb-98dd-40ac2a368769.png">

User Profile (after saving edits)
<img width="891" alt="image" src="https://user-images.githubusercontent.com/65790318/99160693-b77c5880-269e-11eb-9cbd-fba5f950a84c.png">



CC: @erinysong
